src/bitcoin-cli.cpp

```diff
@@ -540,15 +540,15 @@ class NetinfoRequestHandler : public BaseRequestHandler
         // Report detailed peer connections list sorted by direction and minimum ping time.
         if (DetailsRequested() && !m_peers.empty()) {
             std::sort(m_peers.begin(), m_peers.end());
-            result += strprintf("<->   type   net  v  mping   ping send recv  txn  blk  hb %*s%*s%*s ",
-                                m_max_addr_processed_length, "addrp",
-                                m_max_addr_rate_limited_length, "addrl",
-                                m_max_age_length, "age");
+            result += tfm::format_raw("<->   type   net  v  mping   ping send recv  txn  blk  hb %*s%*s%*s ",
```

Comment:
we could avoid formatting these by adding something like:
```C++
if (*it == '*' || ((*it == '_' || *it == '-') && *std::next(it) == '*')) {
    return; // We can't validate format string with %* or %_
}
```
to `Detail_CheckNumFormatSpecifiers` , which would enable us using `tfm::format` here - maybe later we'll implement full validation, but this way at least everything before the `*` format is validated properly.

Reply:
Weren't we already doing partial validation?
Doing a full one for these examples wouldn't be *that* difficult - do you suggest we do that instead?

Reply:
Alternatively, if that's not desired, what if we replaced the dynamic widths with explicit `PadLeft`/`PadRight` calls, e.g.
```C++
result += tfm::format("<->   type   net  v  mping   ping send recv  txn  blk  hb %s%s%s ",
                      PadRight("addrp", m_max_addr_processed_length),
                      PadRight("addrl", m_max_addr_rate_limited_length),
                      PadRight("age", m_max_age_length));
 ```
 and
 ```C++
 result += tfm::format("%s %s%s\n",
                      PadLeft("id", m_max_id_length),
                      IsAddressSelected() ? PadRight("address", m_max_addr_length) : "",
                      IsVersionSelected() ? "version" : "");
```
etc.
I can provide this in a follow-up, if the reviewers think it's a good direction.

<details>
<summary>PadLeft/PadRight</summary>

```C++
[[nodiscard]] inline std::string PadLeft(const std::string_view& str, size_t width, const char pad = ' ')
{
    if (str.size() >= width) return std::string(str);
    return std::string(width - str.size(), pad) + std::string(str);
}

[[nodiscard]] inline std::string PadRight(const std::string_view& str, size_t width, const char pad = ' ')
{
    if (str.size() >= width) return std::string(str);
    return std::string(str) + std::string(width - str.size(), pad);
}
```

</details>

Reply:
Extracted it to https://github.com/bitcoin/bitcoin/pull/30999/files#diff-71badc1cc71ba46244f7841a088251bb294265f4fe9e662c0ad6b15be540eee4R45

Reply:
Resolving since it has its own PR now.